src/django_browser_reload/views.py

```diff
@@ -145,14 +150,37 @@ def events(request: HttpRequest) -> HttpResponseBase:
     if not request.accepts("text/event-stream"):
         return HttpResponse(status=HTTPStatus.NOT_ACCEPTABLE)
 
-    def event_stream() -> Generator[bytes, None, None]:
-        while True:
-            yield message("ping", versionId=version_id)
-
-            should_reload = should_reload_event.wait(timeout=PING_DELAY)
-            if should_reload:
-                should_reload_event.clear()
-                yield message("reload")
+    event_stream: Callable[[], AsyncGenerator[bytes, None]] | Callable[
+        [], Generator[bytes, None, None]
+    ]
+
+    if isinstance(request, ASGIRequest):
+        if django.VERSION < (4, 2):
+            return HttpResponse(
+                "ASGI requires Django 4.2+",
+                status=HTTPStatus.NOT_IMPLEMENTED,
+                content_type="text/plain",
+            )
+
+        async def event_stream() -> AsyncGenerator[bytes, None]:
+            while True:
+                await asyncio.sleep(PING_DELAY)
+                yield message("ping", versionId=version_id)
+
+                if should_reload_event.is_set():
+                    should_reload_event.clear()
+                    yield message("reload")
```

Comment:
I just realized that this is not using an asyncio event, but always sleeping a second between checks. That's imposing an average of 0.5 seconds delay on reloads, which is small but noticeable. Fixing in #178.